src/pybamm/models/full_battery_models/base_battery_model.py

```diff
@@ -18,6 +18,29 @@ def represents_positive_integer(s):
         return val > 0
 
 
+def _ensure_tuple(value: str | tuple[str]) -> tuple | None:
+    """
+    Ensure the input value is returned as a tuple.
+
+    Parameters
+    ----------
+    value : str or tuple
+        The input value to ensure as a tuple.
+
+    Returns
+    -------
+    tuple or None
+        The input value as a tuple, or None if the input is not a string or tuple.
+    """
+    return (
+        (value,)
+        if isinstance(value, str)
+        else value
+        if isinstance(value, tuple)
+        else None
+    )
```

Comment:
Are there cases where `value` can be something other than `str` or `tuple`? If yes, then the type signature of this function is wrong. If not, `None` should not be returned. (It is always preferred to not have the return value of a function depend on an `if` condition, so removing the `None` would be the ideal case.)

Also, I would really prefer a nested `if-else` block here for readability.

Reply:
Ah, just saw @agriyakhetarpal's comment above.

Reply:
The type signature should still be fixed + it should be converted into an `if-else` block.